lib/chef/knife/winrm_knife_base.rb

```diff
@@ -0,0 +1,207 @@
+#
+# Author:: Steven Murawski (<smurawski@chef.io)
+# Copyright:: Copyright (c) 2015 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+require 'chef/knife'
+require 'chef/knife/winrm_base'
+require 'chef/knife/winrm_shared_options'
+
+class Chef
+  class Knife
+    module WinrmCommandSharedFunctions  
+      def self.included(includer)
+        includer.class_eval do
+
+          @@ssl_warning_given = false
+
+          include Chef::Knife::WinrmBase
+          include Chef::Knife::WinrmSharedOptions          
+
+          def configure_session               
+            resolve_session_options
+            resolve_target_nodes       
+            session_from_list
+          end
+
+          def resolve_target_nodes
```

Comment:
this looks like a replacement for Chef::Knife#configure_session. Could you open up an issue against Chef and leave a comment here that says we're duplicating that code until the bug is fixed, at which point we can use that method?